packages/block-library/src/embed/edit.native.js

```diff
@@ -173,15 +173,22 @@ const EmbedEdit = ( props ) => {
 	// Handle incoming preview
 	useEffect( () => {
 		if ( preview && ! isEditingURL ) {
-			// Even though we set attributes that get derived from the preview,
-			// we don't access them directly because for the initial render,
-			// the `setAttributes` call will not have taken effect. If we're
-			// rendering responsive content, setting the responsive classes
-			// after the preview has been rendered can result in unwanted
-			// clipping or scrollbars. The `getAttributesFromPreview` function
-			// that `getMergedAttributes` uses is memoized so that we're not
-			// calculating them on every render.
-			setAttributes( getMergedAttributes() );
+			// When obtaining an incoming preview, we set the attributes derived from
+			// the preview data. In this case, we don't use the `getMergedAttributes`
+			// function because we need to specify an empty classname as the previous
+			// classname might not apply to the new preview.
+			const { allowResponsive } = attributes;
+			setAttributes( {
```

Comment:
Since this logic is similar to the `getMergedAttributes()` couldn't a flag called `ignorePreviousClassnames` be passed and as such the previous classname wouldn't be used and thus eliminating duplicate logic?  Overall, including the logic here is with me :)

Reply:
Great idea! I'll unify this part into the `getMergedAttributes` function 👍.

Reply:
I've just applied this suggestion in [this commit](https://github.com/WordPress/gutenberg/pull/35321/commits/bb6994f3985863b7a9b8f96bbaa598ab16d00692).